Automattic/jetpack
# Modules: Update any relevant module usage of plugins_loaded

Extension of https://github.com/Automattic/jetpack/pull/39841

## Proposed changes:

* This PR changes the hook that the `Upcoming_Events_Shortcode` class hooks into, from `plugins_loaded` to `after_setup_theme`, to ensure the Upcoming Events shortcode works properly
* It also updates a comment to refer to an updated hook name.

### Other information:


## Jetpack product discussion

https://github.com/Automattic/jetpack/pull/39841#issuecomment-2449209410
On the shortcode filter: p1730377751701169-slack-C01U2KGS2PQ (where the comment was changed)

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

To replicate the issue: 


* On a test site, on trunk or using Bleeding Edge if using the Beta tester plugin, ensure that Shortcode embeds are enabled from `wp-admin/admin.php?page=jetpack#/writing`.
* Add an upcoming events shortcode to a post or page - `[upcomingevents]`. No need to add a calendar URL. But if you wnt to test that too - there are some links available here: `https://calagator.org/`
* View the post / page - the shortcode will be unrendered.

To test the fix:
* Apply this PR.
* Visit the same post / page - the shortcode should render (with no Calendar it will show text saying `you must specify a URL to an iCalendar feed in the shortcode. This notice is only displayed to administrators.`.

Test on WoA as well as self-hosted. Simple does not have the same issue (it loads as expected currently), but test that this isn't affected by this PR by following the steps in the generated comment below on a sandboxed site.
